packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -445,6 +445,19 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
     _scrollController.jumpTo(_getScrollOffsetForCaret(caretRect));
   }
 
+  bool _textChangedSinceLastCaretUpdate = false;
+
+  void _handleCaretChanged(Rect caretRect) {
+    // If the caret location has changed due to an update to the text or
+    // selection, then scroll the caret into view.
+    if (_textChangedSinceLastCaretUpdate) {
+      _textChangedSinceLastCaretUpdate = false;
+      scheduleMicrotask(() {
+        _scrollController.jumpTo(_getScrollOffsetForCaret(caretRect));
```

Comment:
(If we want to use jumpTo and not animate, then the approach here is wrong; we should be testing for the caret being out of view during the animation phase, then changing the scroll offset then, so that we never paint it with the overflow.)